gcovr/utils.py

```diff
@@ -139,22 +140,31 @@ def calculate_coverage(covered, total, nan_value=0.0):
     return nan_value if total == 0 else round(100.0 * covered / total, 1)
 
 
-def build_filter(logger, regex):
-    # Try to detect unintended backslashes and warn.
-    # Later, the regex engine may or may not raise a syntax error.
-    # An unintended backslash is a literal backslash r"\\",
-    # or a regex escape that doesn't exist.
-    (suggestion, bs_count) = re.subn(
-        r'\\\\|\\(?=[^\WabfnrtuUvx0-9AbBdDsSwWZ])', '/', regex)
-    if bs_count:
-        logger.warn("filters must use forward slashes as path separators")
-        logger.warn("your filter : {}", regex)
-        logger.warn("did you mean: {}", suggestion)
-
-    if os.path.isabs(regex):
-        return AbsoluteFilter(regex)
-    else:
-        return RelativeFilter(os.getcwd(), regex)
+class FilterOption(object):
+    def __init__(self, regex, path_context=None):
+        if not regex:
+            raise ArgumentTypeError("filter cannot be empty")
```

Comment:
Not sure if `ArgumentTypeError` is the most appropriate exception to use here, but I'd say it's the most straightforward to get behavior that's as similar as possible to before. I evaluated other alternatives but none really stuck. `ValueError` could be used, but using it will print a standardized error rather than a custom one. It will also print a traceback which the Argparse exception handling won't. If the behavior of the Argparse exception is to be preferred (mainly no traceback) a custom program exit function would need to be written. It would of course also be possible to leverage `check_non_empty()` to check that the value is set before passing it to the class constructor similar to how it was before, but then it would only apply to CLI args and not config file filters.